### PR TITLE
Extend generator to support overlays

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
## Describe your changes

Extend model generator to support VSS overlay snippets like handled in VSS tools.
See https://covesa.github.io/vehicle_signal_specification/rule_set/overlay/
and https://github.com/COVESA/vss-tools/blob/master/docs/vspec2x.md 

## Issue ticket number and link

AD#_12603_

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

File: gen_vehicle_model.py [hunk 1]
```diff
@@ -85,17 +103,36 @@
...
+            print(f"Applying VSS overlay from {overlay}...")
+            overlay_tree = vspec.load_tree(
+                overlay,
+                include_dirs,
+                merge_private=False,
+                break_on_unknown_attribute=strict,
+                break_on_name_style_violation=strict,
+                expand_inst=False
+            )
+            vspec.merge_tree(tree, overlay_tree)
```
Comment (dennismeister93) [inline gen_vehicle_model.py:134]: If you want to use the merged tree for model generation i think you missed to reassign the tree from line 114

Comment (dennismeister93) [inline gen_vehicle_model.py:134]:
```suggestion
            tree = vspec.merge_tree(tree, overlay_tree)
```

Comment (BjoernAtBosch) [reply gen_vehicle_model.py:134]: No, `merge_tree` does not return something. It merges the overlay into the base tree (i.e. 1st parameter).

**Review by dennismeister93 — APPROVED** [review]
LGTM! tested and works

**Pull request merged**